File: src/Samples/CSharpApp/ViewModels/MainWindowViewModel.cs
namespace CSharpApp.ViewModels
{
    using System.Collections.ObjectModel;
    using System.Linq;
    using Arcadia.MVVM;
    using Data;
    using Models;

    public class MainWindowViewModel : ViewModelBase
    {
        public MainWindowViewModel() : base()
        {
            // services
            var dataService = new DataService();

            // models
            var simpleCalc = new SimpleCalculationEngine();
            var orderCalc = new OrderCalculationEngine(dataService);

            // page viewmodels
            PageViewModels = new ObservableCollection<PageViewModel>();
            PageViewModels.Add(new SimpleGraphViewModel(simpleCalc));
            PageViewModels.Add(new OrderViewModel(orderCalc));
            PageViewModels.Add(new OrderGraphViewModel(orderCalc));

            CurrentPage = PageViewModels.FirstOrDefault();

            // set order calculation engine to automatic calculation
            orderCalc.Calculation.Automatic = true;
        }

        public PageViewModel CurrentPage { get; set; }

        public ObservableCollection<PageViewModel> PageViewModels { get; private set; }

        public string Title {get { return "Utopia v0.0"; } }
    }
}

File: src/Samples/FSharpApp/ViewModels/MainWindowViewModel.fs
namespace FSharpApp.ViewModels

open System.Collections.ObjectModel
open Arcadia.MVVM
open FSharpApp.Data
open FSharpApp.Models

type MainWindowViewModel() =
    inherit ViewModelBase()

    let pages = new ObservableCollection<PageViewModel>()
    let mutable currentPage = new PageViewModel()

    // services
    let dataService = new DataService()

    // models
    let simpleCalc = new SimpleCalculationEngine()
    let orderCalc = OrderCalculationEngine(dataService)

    // page viewmodels
    let simpleGraphVM = new SimpleGraphViewModel(simpleCalc)
    let orderVM = new OrderViewModel(orderCalc)
    let orderGraphVM = new OrderGraphViewModel(orderCalc)

    let addPage page = pages.Add(page :> PageViewModel)

    do
          simpleGraphVM |> addPage
          orderVM |> addPage
          orderGraphVM |> addPage
          currentPage <- pages.[0]

          // set order calculation engine to automatic calculation
          orderCalc.Calculation.Automatic <- true

    member val CurrentPage = currentPage
   
    member this.Title = "Arcadia v0.2"
    member this.PageViewModels = pages